WFInfo/Login.xaml.cs

```diff
@@ -1,124 +1,159 @@
-                Main.searchBox.IsInUse = true;
-                Main.searchBox.searchField.Focusable = true;
-            }
-        }
-
-        /// <summary>
-        /// Clears the email field if it's not been used yet.
-        /// </summary>
-        /// <param name="sender"></param>
-        /// <param name="e"></param>
-        private void Email_GotFocus(object sender, RoutedEventArgs e)
-        {
-            if (Email.Text == "Email")
-                Email.Text = "";
-        }
-
-        /// <summary>
-        /// Allow the window to be spawned in an appropriate place.
-        /// </summary>
-        /// <param name="x">Left most border of the window</param>
-        /// <param name="y">Top most border of the window</param>
-        public void MoveLogin(double x, double y)
-        {
-            Left = x;
-            Top = y;
-            Show();
-        }
-    }
-}
+﻿using System;
+using System.Runtime.InteropServices;
+using System.Windows;
+using System.Windows.Input;
+using Google.Apis.Auth;
+using Microsoft.Win32;
+using System.Windows.Media;
+
+namespace WFInfo
+{
+    /// <summary>
+    /// Interaction logic for Login.xaml
+    /// </summary>
+    public partial class Login : Window
+    {
+
+        #region default methods
+        public Login()
+        {
+            InitializeComponent();
+        }
+
+        /// <summary>
+        /// Hides the window 
+        /// </summary>
+        /// <param name="sender"></param>
+        /// <param name="e"></param>
+        private void HideExternal(object sender, MouseButtonEventArgs e)
+        {
+            Main.searchBox.Hide(); //hide search it if the user minimizes the window even though not yet logged in
+            Hide();
+        }
+
+        /// <summary>
+        /// Allows the window to be dragged
+        /// </summary>
+        /// <param name="sender"></param>
+        /// <param name="e"></param>
+        private new void MouseDown(object sender, MouseButtonEventArgs e)
+        {
+            if (e.ChangedButton == MouseButton.Left)
+                DragMove();
+        }
+        #endregion
+
+        /// <summary>
+        /// Attempts to log in with the filled in credentials.
+        /// </summary>
+        /// <param name="sender"></param>
+        /// <param name="e"></param>
+        private async void LoginClick(object sender, MouseButtonEventArgs e)
+        {
+
+            if (!Main.dataBase.IsJwtAvailable())
+            {
+                try
+                {
+                    await Main.dataBase.GetUserLogin(Email.Text, Password.Password);
+                }
+                catch (Exception ex)
+                {
+                    Main.dataBase.JWT = null;
+                    Settings.Save();
+                    Main.AddLog("Couldn't login: " + ex);
+                    var StatusMessage = "Status"; //StatusMessage = text to display on StatusUpdate() AND the error box under login 

```

Comment:
Strange initialization, there's always a value written to this so you can just make it a String.empty

Reply:
test forget, was initially only var StatusMessage;